### PR TITLE
feat: Voice search implemented on skills listing

### LINKED ISSUE #2082: Implement voice search in susi
**Describe the feature you'd like**

There should be a search by voice option in skill search activity.

**Additional context**

Make app more handy for user.

**Screenshots**

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2082 

Changes: 
Added voice search icon in action bar, clicking on which prompts voice search.

Screenshots for the change: 
ezgif com-video-to-gif

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt [hunk 1]
```diff
@@ -172,10 +176,36 @@ class SkillsActivity : AppCompatActivity(), SkillFragmentCallback {
...
     }
 
+    //Passes a voice intent, to detect the query that user asks via voice
+    private fun handleVoiceSearch() {
+        val intent = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL, RecognizerIntent.LANGUAGE_MODEL_FREE_FORM)
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, Locale.getDefault())
+
+        if (intent.resolveActivity(packageManager) != null) {
+            startActivityForResult(intent, 10)//Sends the detected query to search
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt]: Declare `10` as constant.

File: app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt [hunk 2]
```diff
@@ -172,10 +176,36 @@ class SkillsActivity : AppCompatActivity(), SkillFragmentCallback {
...
+            startActivityForResult(intent, Constant.VOICE_SEARCH_REQUEST_CODE)//Sends the detected query to search
+        } else {
+            Toast.makeText(this, R.string.error_voice_search, Toast.LENGTH_SHORT)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        when (requestCode) {
+            10 -> if (resultCode == Activity.RESULT_OK && data != null) {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt]: You declared it as constant, and still using directly.

File: app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt [hunk 3]
```diff
@@ -92,4 +92,7 @@ object Constant {
 
     //Visit Website Url
     const val SUSI_VISIT_WEBSITE = "https://chat.susi.ai/"
+
+    //Voice search request code
+    const val VOICE_SEARCH_REQUEST_CODE = 10
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt]: Make it a local constant

**Comment by atm1504** [comment]
@iamareebjamal please review it

**Comment by atm1504** [comment]
@iamareebjamal please review it

File: app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt [hunk 4]
```diff
@@ -176,10 +181,36 @@ class SkillsActivity : AppCompatActivity(), SkillFragmentCallback {
...
+        } else {
+            Toast.makeText(this, R.string.error_voice_search, Toast.LENGTH_SHORT)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        when (requestCode) {
+            VOICE_SEARCH_REQUEST_CODE -> if (resultCode == Activity.RESULT_OK && data != null) {
+                val result = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt]: When and if statements can be combined

**Comment by atm1504** [comment]
@iamareebjamal @liveHarshit please review it.

File: app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt [hunk 5]
```diff
@@ -176,10 +181,34 @@ class SkillsActivity : AppCompatActivity(), SkillFragmentCallback {
...
+        if (intent.resolveActivity(packageManager) != null) {
+            startActivityForResult(intent, VOICE_SEARCH_REQUEST_CODE) // Sends the detected query to search
+        } else {
+            Toast.makeText(this, R.string.error_voice_search, Toast.LENGTH_SHORT)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if ((requestCode == VOICE_SEARCH_REQUEST_CODE) && (resultCode == Activity.RESULT_OK) && data != null) {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt]: Remove extra parentheses.

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt]: Do check now.

**Pull request merged**